code/backend/batch/utilities/helpers/processors/DocumentProcessorHelper.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import List
+
+from ..AzureBlobStorageClient import AzureBlobStorageClient
+
+from ..Processor import Processor
+from ..ConfigHelper import ConfigHelper
+
+from .ProcessorBase import ProcessorBase
+from ..AzureSearchHelper import AzureSearchHelper
+from ..DocumentLoadingHelper import DocumentLoading
+from ..DocumentChunkingHelper import DocumentChunking
+from ...common.SourceDocument import SourceDocument
+
+logger = logging.getLogger(__name__)
+
+
+class DocumentProcessor(ProcessorBase):
+    def __init__(self, blob_client: AzureBlobStorageClient):
+        self.blob_client = blob_client
+
+    def process_file(self, source_url: str, file_name: str):
+        file_extension = file_name.split(".")[-1]
+        processors = [
+            x
+            for x in ConfigHelper.get_active_config_or_default().document_processors
+            if x.document_type.lower() == file_extension.lower()
+        ]
+        self.process(source_url=source_url, processors=processors)
+        if file_extension != "url":
+            self.blob_client.upsert_blob_metadata(
+                file_name, {"embeddings_added": "true"}
+            )
+
+    def process(self, source_url: str, processors: List[Processor]):
```

Comment:
Will this method ever be used outside of this class/file? If not, can we make it private?

Reply:
I realized this is used for url embeddings, so we can leave it as is.